src/utils/format.ts

```diff
@@ -56,17 +56,17 @@ export function sanitizeNegativeAndMakeMultipleOf(value?: string | null, default
     : defaultValue
 }
 
-export function validatePositive(value: string): true | string {
-  return Number(value) > 0 || 'Invalid amount'
+export function validatePositive(value: string, constraint = 0): true | string {
+  return Number(value) > constraint || 'Invalid amount'
 }
 export const validInputPattern = new RegExp(/^\d+\.?\d*$/) // allows leading and trailing zeros
 export const leadingAndTrailingZeros = new RegExp(/(^0*(?=\d)|\.0*$)/, 'g') // removes leading zeros and trailing '.' followed by zeros
 export const trailingZerosAfterDot = new RegExp(/(.*\.\d+?)0*$/) // selects valid input without leading zeros after '.'
 
-export const formatValidity = (validTime: string | number): string =>
-  +validTime == 0
-    ? 'Unlimited'
-    : +validTime < 0
+export const formatValidity = (validTime: string | number, zeroValueText: string, constraint = 0): string =>
```

Comment:
ubernit: shouldn't you be coherent with the naming `constraint` and `zeroValueText `. 
U understand you created `constraint` to allow to use something different than `zero`, but then the label is has `zero` in it.

Also `validTime < constraint` I think there you mean `validTime < 0` for several reasons, one being that you want to be coherent with the message `Invalid time - time cannot be negative`. I think that's unrelated to the `constraint`

Then, `constraint` I don't see you use that anywere. So I think that is using always the default. So for now I would just remove it, otherwise it only makes the code more complicated.